engine/common/requester/engine.go

```diff
@@ -422,6 +436,19 @@ func (e *Engine) onEntityResponse(originID flow.Identifier, res *messages.Entity
 			return fmt.Errorf("could not decode entity: %w", err)
 		}
 
+		if item.checkIntegrity {
+			actualEntityID := entity.ID()
+			// validate that we got correct entity, exactly what we were expecting
+			if entityID != actualEntityID {
+				e.log.Error().
+					Hex("origin", logging.ID(originID)).
+					Hex("stated_entity_id", logging.ID(entityID)).
+					Hex("provided_entity", logging.ID(actualEntityID)).
+					Msg("provided entity does not match stated ID")
+				continue
```

Comment:
So what we are discussing is whether to use `continue` or `return` here. 

if we use `continue`, then we will accept all valid entities and discard the invalid ones (irrespective of order); 

if we use `return`, then we will discard all entities if at least one is invalid.

I'm OK to use continue here, just that better to add comment about this behavior.